File: 프로그래머스/0/120818. 옷가게 할인 받기/옷가게 할인 받기.js
function solution(price) {
    if (price >= 500000) {
        return parseInt(price * 0.8);  // 50만원 이상: 20% 할인
    } else if (price >= 300000) {
        return parseInt(price * 0.9);  // 30만원 이상: 10% 할인
    } else if (price >= 100000) {
        return parseInt(price * 0.95); // 10만원 이상: 5% 할인
    } else {
        return price;        // 10만원 미만: 할인 없음
    }
}

File: 프로그래머스/0/120818. 옷가게 할인 받기/옷가게 할인 받기.kt
class Solution {
    fun solution(price: Int): Int {
        return when {
            price >= 500000 -> price * 8 / 10
            price >= 300000 -> price * 9 / 10
            price >= 100000 -> price * 95 / 100
            else -> price
        }
    }
}